File: MySwitch/MySwitch/ViewController.m
//
//  ViewController.m
//  MySwitch
//
//  Created by Xiangjian Meng on 15/3/12.
//  Copyright (c) 2015年 com.mengxiangjian. All rights reserved.
//

#import "ViewController.h"
#import "MySwitch.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
    MySwitch *mySwitch = [[MySwitch alloc] initWithDrawingPoint:CGPointMake(100, 100)];
    mySwitch.onTitle = @"建议";
    mySwitch.offTitle = @"问题";
    mySwitch.offTitleColor = [UIColor colorWithRed:0.706f green:0.000f blue:0.000f alpha:1.00f];
    mySwitch.onTitleColor = [UIColor whiteColor];
    [self.view addSubview:mySwitch];
    
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOS8/KeychainAuthencation/KeychainAuthencation/ViewController.swift
//
//  ViewController.swift
//  KeychainAuthencation
//
//  Created by Xiangjian Meng on 14-10-8.
//  Copyright (c) 2014年 cn.com.modernmedia. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let keychainManager = KeychainManager();
    var textField : UITextField! = nil;
    var label : UILabel! = nil;

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let viewWidth = self.view.bounds.size.width;
        
        // 添加textField
        textField = UITextField(frame: CGRectMake(20, 50, viewWidth - 40, 30));
        textField.autocapitalizationType = .None;
        textField.autocorrectionType = .No;
        textField.placeholder = "enter your secret";
        textField.borderStyle = .Line;
        textField.textAlignment = NSTextAlignment.Center;
        textField.textColor = UIColor.blackColor();
        self.view.addSubview(textField);
        
        
        // 保存按钮
        let saveButton = UIButton.buttonWithType(.System) as UIButton;
        saveButton.frame = CGRectMake(0, 0, 100, 50);
        saveButton.setTitle("save", forState: .Normal);
        saveButton.center = CGPointMake(viewWidth/2.0, 120);
        saveButton.addTarget(self, action: "save", forControlEvents: UIControlEvents.TouchUpInside);
        self.view.addSubview(saveButton);
        
        // 展示secret的label
        label = UILabel(frame: CGRectMake(0, 200, viewWidth, 50));
        label.textAlignment = .Center;
        label.backgroundColor = UIColor.yellowColor();
        label.textColor = UIColor.blackColor();
        self.view.addSubview(label);

        // 显示secret按钮
        let showSecretButton = UIButton.buttonWithType(.System) as UIButton;
        showSecretButton.frame = CGRectMake(0, 0, 100, 50);
        showSecretButton.center = CGPointMake(viewWidth/2.0, 290);
        showSecretButton.setTitle("show", forState: .Normal);
        showSecretButton.addTarget(self, action: "show", forControlEvents: UIControlEvents.TouchUpInside);
        self.view.addSubview(showSecretButton);
        
    }
    
    func save()
    {
        println("保存\(textField.text)");
        
        keychainManager.savedSecret = textField.text;
    }
    
    func show()
    {
        println("显示密码");
        
        label.text = keychainManager.savedSecret;
    }
    
    

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

